app/src/main/java/com/framgia/fdms/screen/vendor/VendorViewModel.java

```diff
@@ -74,12 +75,59 @@ public void onLoadVendorFailed() {
     }
 
     @Override
-    public void onEditSubmitClick(Vendor vendor) {
+    public void onEditSubmitClick(Producer vendor) {
         // TODO: next task
     }
 
     @Override
-    public void onEditCancelClick(Vendor vendor) {
-        // TODO: next task
+    public void onEditCancelClick(Producer vendor) {
+        mVendorDialog.dismiss();
     }
-}
+
+    protected VendorViewModel(Parcel in) {
+        mPresenter = (VendorContract.Presenter) in.readValue(
+                VendorContract.Presenter.class.getClassLoader());
+        mAdapter = (ListVendorAdapter) in.readValue(ListVendorAdapter.class.getClassLoader());
+        if (in.readByte() == 0x01) {
+            mVendors = new ArrayList<Producer>();
+            in.readList(mVendors, Producer.class.getClassLoader());
+        } else {
+            mVendors = null;
+        }
+        mActivity = (AppCompatActivity) in.readValue(AppCompatActivity.class.getClassLoader());
+        mVendorDialog = (VendorDialog) in.readValue(VendorDialog.class.getClassLoader());
+    }
+
+    @Override
+    public int describeContents() {
+        return 0;
+    }
+
+    @Override
+    public void writeToParcel(Parcel dest, int flags) {
+        dest.writeValue(mPresenter);
+        dest.writeValue(mAdapter);
+        if (mVendors == null) {
+            dest.writeByte((byte) (0x00));
+        } else {
+            dest.writeByte((byte) (0x01));
+            dest.writeList(mVendors);
+        }
+        dest.writeValue(mActivity);
+        dest.writeValue(mVendorDialog);
+    }
+
+    @SuppressWarnings("unused")
+    public static final Parcelable.Creator<VendorViewModel> CREATOR =
+            new Parcelable.Creator<VendorViewModel>() {
+                @Override
+                public VendorViewModel createFromParcel(Parcel in) {
+                    return new VendorViewModel(in);
+                }
+
+                @Override
+                public VendorViewModel[] newArray(int size) {
+                    return new VendorViewModel[size];
+                }
```

Comment:
ERROR: 'method def rcurly' have incorrect indentation level 16, expected level should be one of the following: 8, 12.